thorlcr/thorutil/thbuf.cpp

```diff
@@ -182,7 +182,8 @@ class CSmartRowBuffer: public CSimpleInterface, implements ISmartRowBuffer, impl
                 size32_t left = nb*blocksize-mb.length();
                 memset(mb.reserve(left),0,left);
             }
-            fileio->write(blk*(offset_t)blocksize,mb.length(),mb.bufferBase());
+            tempFileIO->write(blk*(offset_t)blocksize,mb.length(),mb.bufferBase());
+            tmpFileOwner.noteSize(numblocks*blocksize);
```

Comment:
unrelated to this PR, but we're not capturing standard spill (temp) stats from this/these helper functions, i.e. StCycleSpillElapsedCycles, StTimeSpillElapsed, StSizeSpillFile ( NB: we do already in CThorRowCollectorBase )

Is there an existing JIRA? If not can you open one  and link it to this/other JIRA's so it doesn't get lost?

Reply:
Jira for this: https://hpccsystems.atlassian.net/browse/HPCC-31984